### ISSUE TITLE
Local adjustement

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#2538]
Originally reported on Google Code with ID 2555

```
I am working on a simple local control!

Ie that there is only one control area (as Gradient). But I think it's better than
nothing.

If later we arrive at the concept of "layer" or something similar, the algorithms can
be reused

I hope to be able to control in Lab mode : chromaticity, lightness, contrast.

The mouse can be used to delimit the area. I am inspired by the excellent work of Hombre
for "gradient"...

The area will be an "ellipse", with possibility to change width and height (of ellipse).
I also plan a transition adjustable by the user 

:)


```

Reported by `jdesmis` on 2014-11-04 16:17:52

**Comment by Beep6581** [comment]
Patch from Jacques, applies to Gtk+ 2 tip.
https://gist.github.com/anonymous/09b3252448dff3056684
Needs to by run through astyle.

**Comment by Beep6581** [comment]
I applied the code to master and astyled it. Then I put it in its own branch locallab to make merging easier.

Compilation fails with:
```
[ 40%] Building CXX object rtgui/CMakeFiles/rth.dir/editwindow.cc.o
In file included from /home/drslony/repo-rt/rtgui/toolpanelcoord.h:55:0,
                 from /home/drslony/repo-rt/rtgui/batchtoolpanelcoord.h:23,
                 from /home/drslony/repo-rt/rtgui/filepanel.h:23,
                 from /home/drslony/repo-rt/rtgui/editwindow.h:21,
                 from /home/drslony/repo-rt/rtgui/editwindow.cc:18:
/home/drslony/repo-rt/rtgui/locallab.h:50:5: error: ‘Coord’ does not name a type
     Coord draggedCenter;
     ^
rtgui/CMakeFiles/rth.dir/build.make:54: recipe for target 'rtgui/CMakeFiles/rth.dir/editwindow.cc.o' failed
```

**Comment by Hombre57** [comment]
Coord is now in rtengine, so the include should be ../rtengine/coord.h, and the Coord object may have to be prefixed like this:  rtengine::Coord

**Comment by Beep6581** [comment]
Confirmed, works now and is in sync with master-head.
To work on it or just to build it:
`git checkout locallab`

**Comment by Beep6581** [comment]
All locallab issues:
#3546
#3421
#3351 
#3292 
#2538

Merged into #3421
